### PR TITLE
Improve container ready check

### PR DESCRIPTION
**Pull request by xpivarc**
### What this PR does

The existing `ContainerDisksReady` is only checking if the socket exists. This might be in very rare case not enough. Such case is when the container disk container is created later and the handler performs the check right after the socket was created but before it is able to accept a connection.

Fixes #

### Special notes for your reviewer
The only downside I can think of is hiding some degradation or bug in lower part of the stack. Maybe we could expose a metric to track how long it takes for the container disks to be ready.

### Release note

```release-note
Bug-fix: Reduced probability of false "failed to detect socket for containerDisk disk0: ... connection refused" warnings
```

### CODE REVIEW

File: pkg/virt-handler/container-disk/mount.go [hunk 1]
```diff
@@ -365,19 +365,27 @@ func (m *mounter) Unmount(vmi *v1.VirtualMachineInstance) error {
...
 				return false, nil
 			}
+
 		}
 	}
 
 	if util.HasKernelBootContainerImage(vmi) {
-		_, err := m.kernelBootSocketPathGetter(vmi)
+		sock, err := m.kernelBootSocketPathGetter(vmi)
+		if err == nil {
```
Comment (fossedihelm) [inline pkg/virt-handler/container-disk/mount.go:386]: same as above

**Comment by alicefr** [comment]
@xpivarc do we support multiple connections on the socket? Otherwise, when we try with the function it already consumes a connection.

**Comment by alicefr** [comment]
It looks good to me
/lgtm

**Comment by alicefr** [comment]
@xpivarc the unit test failure should be related

File: pkg/virt-handler/container-disk/mount.go [hunk 2]
```diff
@@ -365,19 +365,27 @@ func (m *mounter) Unmount(vmi *v1.VirtualMachineInstance) error {
 func (m *mounter) ContainerDisksReady(vmi *v1.VirtualMachineInstance, notInitializedSince time.Time) (bool, error) {
 	for i, volume := range vmi.Spec.Volumes {
 		if volume.ContainerDisk != nil {
-			_, err := m.socketPathGetter(vmi, i)
+			sock, err := m.socketPathGetter(vmi, i)
```
[the remark this answers is not in the document]
Comment (xpivarc) [reply pkg/virt-handler/container-disk/mount.go:368]: This won't work or am I missing something?

**Comment by xpivarc** [comment]
@alicefr @fossedihelm PTAL
I have added unit tests and made the old ones work

File: pkg/virt-handler/container-disk/mount.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (fossedihelm) [reply pkg/virt-handler/container-disk/mount.go:368]: :man_facepalming: 
Now I see the `err` override...

**Comment by akalenyu** [comment]
/test pull-kubevirt-e2e-k8s-1.29-sig-storage
guest console log pod failed for some reason

**Comment by xpivarc** [comment]
/cherry-pick release-1.4
/cherry-pick release-1.3
Seeing this in the stable branch 1.3

**Pull request merged**